openedx/features/journals/api.py

```diff
@@ -139,24 +163,67 @@ def get_journals(site):
     journals = cache.get(cache_key)
 
     if not journals:
-        catalog_integration = CatalogIntegration.current()
-        if catalog_integration.enabled:
-            try:
-                user = catalog_integration.get_service_user()
-            except ObjectDoesNotExist:
-                return []
-
-            api_client = DiscoveryApiClient(user)
-            journals = api_client.get_journals(orgs)
-            cache.set(cache_key, journals, JOURNALS_CACHE_TIMEOUT)
-        else:
+        api_client = DiscoveryApiClient()
+        if not api_client:
             return []
+        journals = api_client.get_journals(orgs)
+        cache.set(cache_key, journals, JOURNALS_CACHE_TIMEOUT)
 
     return journals
 
 
+def filter_bundles_by_course(bundles, course_id):
+    matching_bundles = []
+    for bundle in bundles:
+        if course_id in [course['key'] for course in bundle['courses']]:
+            matching_bundles.append(bundle)
+    return matching_bundles
+
+
+def get_journal_bundles(site, course_id='', bundle_uuid=''):
+    """Retrieve journal bundles from the discovery service.
+
+    Returns:
+        list of dict, representing journal bundles
+    """
+    if not journals_enabled():
+        return []
+
+    api_resource = 'journal_bundles'
+
+    cache_key = get_cache_key(
+        site_domain=site.domain,
+        resource=api_resource,
+        course_id=course_id,
+        bundle_uuid=bundle_uuid
+    )
+
+    journal_bundles = cache.get(cache_key)
+
+    if not journal_bundles:
+        api_client = DiscoveryApiClient()
+        if not api_client:
+            return []
+        journal_bundles = api_client.get_journal_bundles(uuid=bundle_uuid)
+        if course_id:
+            journal_bundles = filter_bundles_by_course(journal_bundles, course_id)
```

Comment:
wouldn't this filtering better be done on the server side? i.e. add a filter by course_id to the discovery service api for journal bundles